### ISSUE TITLE
Skip certain folders and/or files from alt text generation

### ISSUE THREAD

**Issue by zizther** [alttextlab/alt-text-craftcms#6]
Hi.

Is it possible to skip certain folders and/or files through a regex, so the automation or bulk generation would skip these images.

My use-case is we have a large number of images which are imported from a 3rd party, not all images require the alt text to be generated, this would save credit on the account so it isn't used unnecessarily.

Thanks

**Comment by andreizanik** [comment]
Hi @zizther

We’re in the final stages of implementing the requested feature.

Here’s what we’ve done: a new **Disable Volumes** setting will let you select the asset volumes you want to ignore.


We chose this approach because a single "filesystem" can be linked to one "asset volume" when no subpath is specified in the volume settings, or the same "filesystem" can be linked to multiple "asset volumes" when a subpath is specified.

As a result, we can now control which volumes and subfolders should have alt text generated and which ones should be skipped.

Once testing is complete, we’ll release a new version of the plugin.

I’d love to hear your thoughts on this solution.
Thanks for the feature request!

**Comment by zizther** [comment]
@andreizanik Thanks for this. It works well.
Do you know if it would be possible to support subfolders within a volume as well.

Our use-case if we tend to use a single volume, but want to exclude subfolders within the volume

**Comment by zizther** [comment]
@andreizanik

In our use case we have a single filesystem and single volume volume.
Within that I have a folder of employees which I don't need alt text generated for.

An example of the folder structure within the single Asset Volume could be:

```
|- /images
|--- /people
|--- /quotes
|--- /file1.jpg
|--- /file2.jpg
```

In this example I would want to somehow exclude the `people` directory.

Where you currently have the ability to exclude certain volumes, it would be good to have an option which would ignore a certain folder from that volume, rather than the whole volume.
I presume this could be done via typing the folder name in a field, as it may not be possible to select it from the volume.

I understand if this requirement could be too granular and hard to manage.
Would be good to get your thoughts.

**Comment by andreizanik** [comment]
It seems to me that the more “correct” approach might be to create several separate Asset Volumes for different needs. That said, I’m not claiming to know the full specifics of your project, so I may be wrong here.

In any case, this feels like a very specific use case, and we’d prefer not to complicate the plugin settings for all users because of it. Most users likely wouldn’t need this level of granularity.

That said, we’ll keep thinking about your request.
One initial idea that came to mind is using an environment variable with a regular expression to check the file path.

**Comment by andreizanik** [comment]
Hi @zizther, 
1.3.0 version has been published and now you can use regular expression for exclude images.
 
We updated our documentation 
https://github.com/alttextlab/alt-text-craftcms/tree/v1.3.0?tab=readme-ov-file#method-2-advanced-exclusion-via-regex-environment-variable
